UserService/app/src/main/java/com/bookmyshow/userservice/exception/handler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,41 @@
+package com.bookmyshow.userservice.exception.handler;
+
+import com.bookmyshow.userservice.exception.PermissionNotFoundException;
+import com.bookmyshow.userservice.exception.RoleNotFoundException;
+import com.bookmyshow.userservice.exception.UserNotFoundException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.ServletWebRequest;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler({RoleNotFoundException.class, PermissionNotFoundException.class, UserNotFoundException.class})
+    public ResponseEntity<ErrorDetails> handleNotFoundException(RuntimeException exception, WebRequest request){
+        return new ResponseEntity<>(new ErrorDetails(exception.getMessage(), ((ServletWebRequest) request).getRequest().getRequestURI()), HttpStatus.NOT_FOUND);
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(MethodArgumentNotValidException ex, HttpHeaders headers, HttpStatus status, WebRequest request) {
+
+        Map<String, String> errorMap = ex.getBindingResult().getFieldErrors().stream()
+                .collect(Collectors.toMap(FieldError::getField, fieldError -> {
+                    String message = fieldError.getDefaultMessage();
+                    if(message!=null) return message;
+                    else return "";
+                }));
+
+        return new ResponseEntity<>(new ErrorDetails(ex.getMessage(), ((ServletWebRequest) request).getRequest().getRequestURI(), errorMap), HttpStatus.BAD_REQUEST);
+    }
```

Comment:
Let us mark this as a Default Base Exception Handler and move it to a core module in the application. We will have a separate ExceptionHandler only for this user service so that it inherits the base handler. This way we will route only custom exceptions only if it is specific to this module else will continue to use base exception handler